### DISCUSSION TITLE
How to add "cache-control: no-cache" and "etag: xxx" headers to my ssr-pages

### DISCUSSION THREAD

**Discussion by aozimuk** [nuxt/nuxt#25937] [Questions]
Hi guys!
I would like to add the "cache-control: no-cache" and "etag: ...." headers to my ssr-pages of my blog (home, articles, etc).
But I can't figure out how to set up RouteRules to do this.
Is there any way to do this using RouteRules? Or should I create a cachedEventHandler and manually calculate the hash for the etag?
Maybe do you have any examples?
I'm not a web developer so I need your help..

Thank you very much!

**Comment by xjccc** [comment]
[legacy-handler-or-middleware](https://nuxt.com/docs/guide/directory-structure/server#legacy-handler-or-middleware)

You can try this.

```
export default fromNodeMiddleware((req, res, next) => {
   res.setHeader('cache-control', 'no-cache')
   next()
})
```

[routeRules](https://nitro.unjs.io/config#routerules)
